tests/regression_tests/mg_temperature/build_2g.py

```diff
@@ -0,0 +1,291 @@
+
+    # roll axis for scatter matrix
+    for t in xs_micro:
+        for n in xs_micro[t]['scatter']:
+            xs_micro[t]['scatter'][n] = np.rollaxis(xs_micro[t]['scatter'][n],
+                                                    0, 3)
+    return xs_micro
+
+
+def create_macro_dict(xs_micro):
+    """Create a dictionary with two group cross-section"""
+    xs_macro = {}
+    for t, d1 in xs_micro.items():
+        xs_macro[t] = {}
+        for r, d2 in d1.items():
+            temp = []
+            xs_macro[t][r] = {}
+            for n, v in d2.items():
+                temp.append(d2[n])
+            xs_macro[t][r]['macro'] = sum(temp)
+    return xs_macro
+
+
+def create_openmc_2mg_libs(names):
+    """Built a micro/macro two group openmc MGXS libraries"""
+    # Initialized library params
+    group_edges = [0.0, 0.625, 20.0e6]
+    groups = openmc.mgxs.EnergyGroups(group_edges=group_edges)
+    mg_cross_sections_file_micro = openmc.MGXSLibrary(groups)
+    mg_cross_sections_file_macro = openmc.MGXSLibrary(groups)
+    # Building a micro mg library
+    micro_cs = create_micro_xs_dict()
+    for name in names:
+        mg_cross_sections_file_micro.add_xsdata(build_openmc_xs_lib(name,
+                                                                    groups,
+                                                                    [t for t in
+                                                                     micro_cs],
+                                                                    micro_cs))
+    # Building a macro mg library
+    macro_xs = create_macro_dict(micro_cs)
+    mg_cross_sections_file_macro.add_xsdata(build_openmc_xs_lib('macro',
+                                                                groups,
+                                                                [t for t in
+                                                                 macro_xs],
+                                                                macro_xs))
+    # Exporting library to hdf5 files
+    mg_cross_sections_file_micro.export_to_hdf5('micro_2g.h5')
+    mg_cross_sections_file_macro.export_to_hdf5('macro_2g.h5')
+    # Returning the macro_xs dict is needed for analytical solution
+    return macro_xs
+
+
+def analytical_solution_2g_therm(xsmin, xsmax=None, wgt=1.0):
+    """ Calculate eigenvalue based on analytical solution for eq Lf = (1/k)Qf
+        in two group for infinity dilution media in assumption of group
+        boundary in thernmal spectra < 1.e+3 Ev
+        Parametres:
+        ----------
+        xsmin : dict
+            - macro cross-sections dictonary with minimum range temperature
+        xsmax : dict
+            - macro cross-sections dictonary with maximum range temperature
+              by default: None not used for standalone temperature
+        wgt : double
+            - weight for interpolation by default 1.0
+        Returns:

```

Comment:
Add an extra line before "Returns" and make sure the "-----------" has the same number of dashes as Returns has spaces (7).